### PR TITLE
Initial work for char server to C++

### PR DESCRIPTION
**Pull request by lighta**
Same kind of work of mapserver but for charserver.

### CODE REVIEW

**Comment by Jeybla** [comment]
Looks straight forward! Thank you lighta, I'll do some tests this evening.

**Comment by Questune09** [comment]
please double check also when get the latest commit all mvp monsters inside lhz nightmare dont attack players they just ignore them thank you

**Comment by Questune09** [comment]
as in all aggresive monsters doesnt attack

**Comment by lighta** [comment]
Alright, @Questune09 but any issues on this commit ?

**Comment by Atemo** [comment]
@Questune09 Aggressives monsters attack my character, I can't reproduce in https://github.com/rathena/rathena/commit/e7e9cf6d7333345d0e11d9f0bf9f1badb6ff3cd9
Tested vs ra_san01, lhz_dun02 and lhz_dun03 mobs.

**Comment by Questune09** [comment]
@everyone nothing i also check gef_dun02 and glast heim dungeons some raydrics and deviruchis doesnt attack also

**Comment by Questune09** [comment]
i also dont use @battleignore command also tested with normal characters

i also provide screen shots

image

**Comment by Jeybla** [comment]
@Questune09 please continue in the related issue: #2572
Can you give us more detailed informations?
Your operating system, your git hash, your `g++ --version`. Thanks in advance!
(Note: I'll remove the issue related comments from this pull request soon)

**Comment by Questune09** [comment]
im sorry i use

g++ (GCC) 4.4.7 20120313 (Red Hat 4.4.7-18)

File: src/char/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ set( CHAR_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}  CACHE INTERNAL "" )
 #
 if( BUILD_SERVERS )
 message( STATUS "Creating target char-server" )
-file(GLOB CHAR_HEADERS ${CHAR_SOURCE_DIR}/*.h)
+file(GLOB CHAR_HEADERS ${CHAR_SOURCE_DIR}/*.hpp)
 file(GLOB CHAR_SOURCES_C ${CHAR_SOURCE_DIR}/*.c)
```
Comment (Lemongrass3110) [inline src/char/CMakeLists.txt]: Could be removed?

File: src/char/char_mapif.cpp [hunk 2]
```diff
@@ -1132,10 +1133,10 @@ int chmapif_parse_updfamelist(int fd){
...
             switch(type)
             {
-				case RANK_BLACKSMITH:	size = fame_list_size_smith;	list = smith_fame_list;		break;
-				case RANK_ALCHEMIST:	size = fame_list_size_chemist;	list = chemist_fame_list;	break;
-				case RANK_TAEKWON:		size = fame_list_size_taekwon;	list = taekwon_fame_list;	break;
-				default:				size = 0;						list = NULL;				break;
+            case RANK_BLACKSMITH:	size = fame_list_size_smith;	list = smith_fame_list;		break;
+            case RANK_ALCHEMIST:	size = fame_list_size_chemist;	list = chemist_fame_list;	break;
+            case RANK_TAEKWON:		size = fame_list_size_taekwon;	list = taekwon_fame_list;	break;
+            default:				size = 0;						list = NULL;				break;
```
Comment (Lemongrass3110) [inline src/char/char_mapif.cpp]: Style change only? Could be reverted for now?

File: src/char/int_pet.hpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Athena Dev Teams - Licensed under GNU GPL
+// For more information, see LICENCE in the main folder
+
+#ifndef _INT_PET_SQL_H_
+#define _INT_PET_SQL_H_
+
+#include "../common/cbasetypes.h"
+
+  struct s_pet;
```
Comment (Lemongrass3110) [inline src/char/int_pet.hpp]: A lot of spaces at the beginning of the lines here as well.

File: src/char/char_mapif.cpp (hunk 2 above)
Comment (Jeybla) [reply src/char/char_mapif.cpp]: The whole function is mixed tab and space indentation. Anyway I'll revert it. We should keep style changes in explicit commits.

**Pull request merged**